d20/tests/test_Registration.py

```diff
@@ -177,3 +181,433 @@ def test2registerverification(self):
+        )
+        class RegTestScreen(ScreenTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def present(self):
+                pass
+
+            def filter(self):
+                pass
+
+            def formatData(self, data):
+                pass
+
+    assert str(excinfo.value) == "'options must be of type 'Arguments'"
+
+
+def testScreenRegistrationFormWrongArgument():
+    with pytest.raises(TypeError) as excinfo:
+        @registerScreen(
+            name="ScreenTester2",
+            version="0.1",
+            engine_version="0.1",
+            test="test",
+        )
+        class RegTestScreen(ScreenTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def present(self):
+                pass
+
+            def filter(self):
+                pass
+
+            def formatData(self, data):
+                pass
+
+    assert str(excinfo.value) == "test is an invalid keyword argument"
+
+
+def testBackStoryRegistrationFormWrongOption():
+    with pytest.raises(TypeError) as excinfo:
+        @registerBackStory(
+            name="BackStoryTester1",
+            description="Test BackStory",
+            creator="",
+            version="0.1",
+            engine_version="0.1",
+            category="test",
+            options="abc",
+        )
+        class RegTestBackStory(BackStoryTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def handleData(self, **kwargs):
+                pass
+
+    assert str(excinfo.value) == "'options must be of type 'Arguments'"
+
+
+def testBackStoryRegistrationFormWrongArgument():
+    with pytest.raises(TypeError) as excinfo:
+        @registerBackStory(
+            name="BackStoryTester2",
+            description="Test BackStory",
+            creator="",
+            version="0.1",
+            engine_version="0.1",
+            category="test",
+            test="test",
+        )
+        class RegTestBackStory(BackStoryTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def handleData(self, **kwargs):
+                pass
+
+    assert str(excinfo.value) == "test is an invalid keyword argument"
+
+
+def testBackStoryRegistrationFormNonIterableInterest():
+    with pytest.raises(TypeError) as excinfo:
+        @registerBackStory(
+            name="BackStoryTester2",
+            description="Test BackStory",
+            creator="",
+            version="0.1",
+            engine_version="0.1",
+            category="test",
+            interests=1
+        )
+        class RegTestBackStory(BackStoryTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def handleData(self, **kwargs):
+                pass
+
+    assert str(excinfo.value) == "Expected an interable type"
```

Comment:
we may want to change this to `iterable` and include a fix for the spelling in the code with the PR